src/main/webapp/app/exercises/programming/shared/code-editor/monaco/code-editor-monaco.component.ts

```diff
@@ -0,0 +1,217 @@
+import { Component, EventEmitter, Input, OnChanges, Output, QueryList, SimpleChanges, ViewChild, ViewChildren, ViewEncapsulation } from '@angular/core';
+import { RepositoryFileService } from 'app/exercises/shared/result/repository.service';
+import { CodeEditorRepositoryFileService } from 'app/exercises/programming/shared/code-editor/service/code-editor-repository.service';
+import { CodeEditorFileService } from 'app/exercises/programming/shared/code-editor/service/code-editor-file.service';
+import { LocalStorageService } from 'ngx-webstorage';
+import { MonacoEditorComponent } from 'app/shared/monaco-editor/monaco-editor.component';
+import { firstValueFrom } from 'rxjs';
+import { Annotation, FileSession } from 'app/exercises/programming/shared/code-editor/ace/code-editor-ace.component';
+import { Feedback } from 'app/entities/feedback.model';
+import { Course } from 'app/entities/course.model';
+import { CodeEditorTutorAssessmentInlineFeedbackComponent } from 'app/exercises/programming/assess/code-editor-tutor-assessment-inline-feedback.component';
+import {
+    CommitState,
+    CreateFileChange,
+    DeleteFileChange,
+    EditorState,
+    FileChange,
+    FileType,
+    RenameFileChange,
+} from 'app/exercises/programming/shared/code-editor/model/code-editor.model';
+import { fromPairs, pickBy } from 'lodash-es';
+
+@Component({
+    selector: 'jhi-code-editor-monaco',
+    templateUrl: './code-editor-monaco.component.html',
+    styleUrls: ['./code-editor-monaco.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    providers: [RepositoryFileService],
+})
+export class CodeEditorMonacoComponent implements OnChanges {
+    @ViewChild('editor', { static: true })
+    editor: MonacoEditorComponent;
+    @ViewChildren(CodeEditorTutorAssessmentInlineFeedbackComponent)
+    inlineFeedbackComponents: QueryList<CodeEditorTutorAssessmentInlineFeedbackComponent>;
+    @Input()
+    commitState: CommitState;
+    @Input()
+    editorState: EditorState;
+    @Input()
+    course?: Course;
+    @Input()
+    feedbacks: Feedback[] = [];
+    @Input()
+    isTutorAssessment: boolean = false;
+    @Input()
+    selectedFile: string | undefined = undefined;
```

Comment:
```suggestion
    selectedFile?: string;
```

Reply:
I changed this and two other occurrences 👍